quest/include/precision.h

```diff
@@ -5,7 +5,7 @@
 #ifndef PRECISION_H
 #define PRECISION_H
 
-#include "quest/include/modes.h"
+#include "modes.h"
```

Comment:
I'm still convinced we should follow [Kolpackov's paths](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p1204r0.html) (ranted about [here](https://github.com/QuEST-Kit/QuEST/pull/430#issuecomment-2220524094)), in which case all the changes to `#include "..."` of this PR would be redundant.

Reply:
Oops, I actually thought I'd changed these back, but evidently hallucinated it 😅 Fixing now!